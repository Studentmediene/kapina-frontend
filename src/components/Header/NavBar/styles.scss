@import 'variables.scss';

.navbar {
  flex: 1;
}

.navbarList {
  display: none;
  flex-direction: row;
  justify-content: center;

  padding: 0;
  font-size: 1.3em;
}

.navbarItem {
  list-style: none;
  display: inline;
  text-align: center;
  padding-left: 1.5em;
}

.navbarItem:first-child {
  padding-left: 0;
}

.navbarItem:hover {
  text-decoration: underline;
}

.navbarLink {
  text-decoration: none;
  color: inherit;
}

/* Vertical navbar for very small displays */
@media screen and (min-width: $breakpoint-large) {
  .navbarList {
    display: flex;
  }
}
