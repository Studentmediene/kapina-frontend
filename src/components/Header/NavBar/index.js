import React from 'react';
import { Link } from 'react-router-dom';

import styles from './styles.css';

const navbarLinks = [
  {
    path: '/',
    title: 'Hjem',
  },
  {
    path: '/programmer',
    title: 'Programmer',
  },
  {
    path: '/sendeplan',
    title: 'Sendeplan',
  },
];

const NavBar = () => (
  <ul className={styles.navbarList}>
    {navbarLinks.map(link => (
      <li key={link.path} className={styles.navbarItem}>
        <Link className={styles.navbarLink} to={link.path}>
          {link.title}
        </Link>
      </li>
    ))}
  </ul>
);

export default NavBar;
