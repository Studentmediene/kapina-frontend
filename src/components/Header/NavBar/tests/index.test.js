import React from 'react';
import { shallow } from 'enzyme';

import NavBar from '../';

describe('<NavBar />', () => {
  it('renders correctly', () => {
    const tree = shallow(<NavBar />);
    expect(tree).toMatchSnapshot();
  });
});
