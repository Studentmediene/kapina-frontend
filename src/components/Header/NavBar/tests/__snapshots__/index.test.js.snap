// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<ul
  className="navbarList"
>
  <li
    className="navbarItem"
    key="/"
  >
    <Link
      className="navbarLink"
      replace={false}
      to="/"
    >
      Hjem
    </Link>
  </li>
  <li
    className="navbarItem"
    key="/programmer"
  >
    <Link
      className="navbarLink"
      replace={false}
      to="/programmer"
    >
      Programmer
    </Link>
  </li>
</ul>
`;
